reana_workflow_controller/workflow_run_manager.py

```diff
@@ -393,28 +399,49 @@ def stop_interactive_session(self, interactive_session_id):
                 current_db_sessions.delete(int_session)
                 current_db_sessions.commit()
 
+    def _delete_k8s_job_quiet(self, job_name):
+        """Delete a Kubernetes job.
+
+        This method will not raise an exception if the deletion fails, but will
+        only log the error.
+
+        :param job_name: Name of the Kubernetes job to be deleted.
+        :type job_name: str
+        :return: True if the job was deleted successfully, False otherwise.
+        """
+        try:
+            current_k8s_batchv1_api_client.delete_namespaced_job(
+                job_name,
+                REANA_RUNTIME_KUBERNETES_NAMESPACE,
+                body=V1DeleteOptions(
+                    grace_period_seconds=0, propagation_policy="Background"
+                ),
+            )
+        except ApiException:
+            logging.error(
+                f"Error while trying to stop {self.workflow.id_}"
+                f": Kubernetes job {job_name} could not be deleted.",
+                exc_info=True,
+            )
+            return False
+        return True
+
     def stop_batch_workflow_run(self):
         """Stop a batch workflow run along with all its dependent jobs."""
+        jobs_to_delete = self.get_workflow_running_jobs()
+
+        for job in jobs_to_delete:
+            job_id = job.backend_job_id
+            if self._delete_k8s_job_quiet(job_id):
+                job.status = JobStatus.stopped
+                Session.add(job)
+
+        # Commit the session once all the jobs have been processed.
+        Session.commit()
```

Comment:
Since the `Session` object is created with `autocommit=False` in `reana_db.database`, I thought an explicit commit was needed to persist the changes in the database (as is done in other parts of the code in the same module (such as [here](https://github.com/reanahub/reana-workflow-controller/blob/3125ca819ce5005c97dfbf5c19936e60781622d4/reana_workflow_controller/workflow_run_manager.py#L337) or [here](https://github.com/reanahub/reana-workflow-controller/blob/3125ca819ce5005c97dfbf5c19936e60781622d4/reana_workflow_controller/workflow_run_manager.py#L394)). Even after removing this line apparently the changes are still saved on the database (maybe the session gets committed somewhere else?) but I guess an explicit `commit()` is helpful to make sure that the DB is indeed updated.